.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,199 @@
+        id: cached-poetry-dependencies
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
+      #----------------------------------------------
+      # install dependencies if cache does not exist
+      #----------------------------------------------
+      - name: Install dependencies
+        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: poetry install --with benchmark --without dev
+
+      - name: Setup variables I
+        id: branch1
+        run: |
+          echo ${{ github.event.issue.pull_request }}
+          echo "::set-output name=filepath::benchmark/${{ env.main }}.json"
+          echo "::set-output name=hash::${{ env.main }}"
+
+      - name: Download Testing File
+        run: |
+          curl https://storage.courtlistener.com/bulk-data/eyecite/tests/one-percent.csv.bz2 --output benchmark/bulk-file.csv.bz2
+
+      - name: Run first benchmark
+        run: |
+          poetry run python benchmark/benchmark.py --branches ${{ steps.branch1.outputs.hash }}
+          git stash --include-untracked
+
+      - uses: actions/checkout@v3
+        with:
+          repository: freelawproject/eyecite
+          ref: main
+
+      - name: Load cached venv 2
+        id: cached-poetry-dependencies2
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
+      #----------------------------------------------
+      # install dependencies if cache does not exist
+      #----------------------------------------------
+      - name: Install dependencies 2
+        if: steps.cached-poetry-dependencies2.outputs.cache-hit != 'true'
+        run: poetry install --with benchmark --without dev
+      - name: Setup variables II
+        id: branch2
+        run: |
+          echo "::set-output name=filepath::benchmark/${{ env.main }}.json"
+          echo "::set-output name=hash::${{ env.main }}"
+
+      - name: Run second benchmark
+        run: |
+          git stash pop
+          poetry run python benchmark/benchmark.py --branches ${{ steps.branch1.outputs.hash }} ${{ steps.branch2.outputs.hash }} --pr ${{ github.event.number }}
+          mkdir results
+          mv benchmark/output.csv benchmark/${{ steps.branch1.outputs.hash }}.json benchmark/${{ steps.branch2.outputs.hash }}.json benchmark/report.md benchmark/chart.png results/
+
+      #----------------------------------------------
+      #             Upload to Github PR
+      #----------------------------------------------
+      - name: Pushes test file
+        uses: dmnemec/copy_file_to_another_repo_action@main
+        env:
+          API_TOKEN_GITHUB: ${{ secrets.TOKEN }}
+        with:
+          user_email: 'bill@free.law'
+          user_name: 'flooie'
```

Comment:
This is a bit weird. Is this required? If so, I guess we have a free law bot Github account, I can resurrect.

Reply:
OK, you can use this:

```
user_name: "freelawbot"
user_email: "info@free.law"
```

Reply:
we also - need to personalize and name the correct secret level to enable some of these features.

Reply:
updated